compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -1665,6 +1665,39 @@ void QuantizeWithMinMaxPass::set_output_type(loco::Graph *g) const
   }
 }
 
+/**
+ * How QuantizeWithMinMax works?
+ *
+ * We categorized tensors into four groups
+ * - Activation: Feature maps (both Const/Non-const)
+ * - Weights: Const tensors of specific Ops (Conv, FC, ...)
+ * - Bias: Const tensors of specific Ops (Conv, FC, ...)
+ * - Others: padding value, one_hot value, axis, ..
+ *
+ * Activation is quantized in different ways
+ * 1. For non-constant activation, quantize using recorded min/max
+ * 2. For constant activation, quantize using min/max of its value
+ * 3. For some Ops (ex: pad_v2), output qparam is used as input qparam (backward propagation)
+ * 4. For some Ops (ex: reshape), input qparam is used as output qparam (forward propagation)
+ * 5. For some Ops (ex: tanh), output qparam has pre-defined values
```

Comment:
Just for curiosity!

In following example, which one should be quantized first? Reshape or PadV2?
```
PadV2 --- activation --- Reshape
```

Reply:
In our implementation, PadV2 -> Reshape.

1. PadV2's ofm is quantized in `QuantizeActivation` (using recorded min/max values)
2. qparam of PadV2's ofm is propagated to its input (`propagate_pad_v2_quantparam`) - backward prop
3. qparam of Reshape's ifm is propagated to its output (`PropagateQuantParamPass`) - forward prop

Reshape -> PadV2 would be also ok. as there is no conflict here.

A problematic situation would be Reshape - PadV2. The ofm of reshape (= ifm of PadV2) is quantized as below.
1. Quantize using recorded min/max in `QuantizeActivation`
2. Overwrite qparam of PadV2's ofm - backward prop
3. Overwrite qparam of Reshape's ifm - forward prop

Since we perform backward prop and then forward prop, the result of backward prop (step 2) can be overwritten by forward propagation (step 3).

I'm not sure the order (backward prop ->forward prop) is always safe for all Ops (that's why I did not explicitly write the order in the code). But for now, Ops for forward prop are much more common than Ops for backward prop, so I decided to do backward prop first.